applications/FluidDynamicsApplication/custom_utilities/rotating_frame_utility.cpp

```diff
@@ -0,0 +1,120 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Sebastian Ares de Parga Regalado
+//
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Application includes
+#include "rotating_frame_utility.h"
+
+namespace Kratos 
+{
+
+void RotatingFrameUtility::ApplyVelocityToRotatingObject(
+    ModelPart& rModelPart,
+    const array_1d<double, 3>& rAxisOfRotation, 
+    const double& rOmega, 
+    const array_1d<double, 3>& rCenterOfRotation)
+{
+    KRATOS_TRY;
+
+    // Creating the angular velocity vector
+    array_1d<double, 3> angular_velocity_vector = rOmega * rAxisOfRotation;
+
+    thread_local array_1d<double, 3> position_vector;
```

Comment:
please do not use the thread_local keyword. 

in particular this is a stack array, so you can just define it there where you use it